src/tasks/DockerComposeTaskProvider.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { Task } from 'vscode';
+import { isNewContextType } from '../docker/Contexts';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { cloneObject } from '../utils/cloneObject';
+import { CommandLineBuilder } from '../utils/commandLineBuilder';
+import { resolveVariables } from '../utils/resolveVariables';
+import { DockerComposeOptions, DockerComposeTaskDefinitionBase } from './DockerComposeTaskDefinitionBase';
+import { DockerTaskProvider } from './DockerTaskProvider';
+import { DockerComposeTaskContext, throwIfCancellationRequested } from './TaskHelper';
+
+// This doesn't need to be extended so redefining for parity is simplest
+export type DockerComposeTaskDefinition = DockerComposeTaskDefinitionBase;
+
+export interface DockerComposeTask extends Task {
+    definition: DockerComposeTaskDefinition;
+}
+
+export class DockerComposeTaskProvider extends DockerTaskProvider {
+    public constructor() { super('docker-build', undefined) }
+
+    protected async executeTaskInternal(context: DockerComposeTaskContext, task: DockerComposeTask): Promise<void> {
+        const definition = cloneObject(task.definition);
+        definition.dockerCompose = definition.dockerCompose || {};
+        definition.dockerCompose.files = definition.dockerCompose.files || [];
+
+        await this.validateResolvedDefinition(context, definition.dockerCompose);
+
+        const commandLine = await this.resolveCommandLine(definition.dockerCompose);
+
+        // Because BuildKit outputs everything to stderr, we will not treat output there as a failure
+        await context.terminal.executeCommandInTerminal(
+            commandLine,
+            context.folder,
+            false, // rejectOnStderr
+            undefined, // stdoutBuffer
+            Buffer.alloc(10 * 1024), // stderrBuffer
+            context.cancellationToken
+        );
+        throwIfCancellationRequested(context);
+    }
+
+    private async validateResolvedDefinition(context: DockerComposeTaskContext, dockerCompose: DockerComposeOptions): Promise<void> {
+        if (!dockerCompose.up && !dockerCompose.down) {

```

Comment:
What if both are present?

Reply:
I can add extra validation here. The tasks.json will already have some annoying yellow squiggles if you add both.